aspnetcore/security/anti-request-forgery.md

```diff
@@ -1,18 +1,311 @@
+
+<!-- literal_block {"xml:space": "preserve", "language": "html", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```html
+<form asp-controller="Manage" asp-action="ChangePassword" 
+  method="post" asp-antiforgery="false">
+
+</form>
+```
+
+ASP.NET Core includes three [filters](../mvc/controllers/filters.md) for working with antiforgery tokens: ``ValidateAntiForgeryToken``, ``AutoValidateAntiforgeryToken``, and ``IgnoreAntiforgeryToken``.
+
+### ValidateAntiForgeryToken
+
+The ``ValidateAntiForgeryToken`` is an action filter that can be applied to an individual action, a controller, or globally for the app. Requests made to actions that have this filter applied will be blocked unless the request includes a valid antiforgery token.
+
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```c#
+[HttpPost]
+[ValidateAntiForgeryToken]
+public async Task<IActionResult> RemoveLogin(RemoveLoginViewModel account)
+{
+    ManageMessageId? message = ManageMessageId.Error;
+    var user = await GetCurrentUserAsync();
+    if (user != null)
+    {
+        var result = await _userManager.RemoveLoginAsync(user, account.LoginProvider, account.ProviderKey);
+        if (result.Succeeded)
+        {
+            await _signInManager.SignInAsync(user, isPersistent: false);
+            message = ManageMessageId.RemoveLoginSuccess;
+        }
+    }
+    return RedirectToAction(nameof(ManageLogins), new { Message = message });
+}
+```
+
+The ``ValidateAntiForgeryToken`` attribute will require a token for every request, including *GET* requests. If you apply it broadly, you can override it with the ``IgnoreAntiforgeryToken`` attribute.
+
+### AutoValidateAntiforgeryToken
+
+In most cases, your application will not receive antiforgery tokens for certain kinds of HTTP requests, such as GET requests. Instead of broadly applying the ``ValidateAntiForgeryToken`` attribute and then overriding it with ``IgnoreAntiforgeryToken`` attributes, you can use the ``AutoValidateAntiforgeryToken`` attribute. This attribute works identically to the ``ValidateAntiForgeryToken`` attribute, except that it ignores the following HTTP methods:
+
+* GET
+* HEAD
+* OPTIONS
+* TRACE
+
+The ASP.NET Core team recommends you use ``AutoValidateAntiforgeryToken`` broadly for non-API scenarios, so that your POST actions are protected by default (APIs don't have an automatic mechanism for sending the non-cookie part of the token). The alternative is to ignore antiforgery tokens by default, unless ``ValidateAntiForgeryToken`` is applied to the individual action method. It's more likely in this scenario for a POST action method to be left unprotected, leaving your app vulnerable to CSRF attacks. Even anonymous POSTS should send the antiforgery token.
```

Comment:
>The alternative is to ignore antiforgery tokens by default, unless ``ValidateAntiForgeryToken`` is applied to the individual action method.
An alternative to applying ``AutoValidateAntiforgeryToken`` globally is to apply the ``ValidateAntiForgeryToken`` to each action or class you need to protect. This approach is considered less secure, as a method  or call could be added with the ``ValidateAntiForgeryToken``.

>Even anonymous POSTS should send the antiforgery token.
Good info. I suppose you can tack in onto the end but it doesn't fit great.